### PR TITLE
add spdlog/1.5.0

### LINKED ISSUE #671: [request] spdlog/1.5.0
### Package Details
  * Package Name/Version: **spdlog/3.15.3**
  * Changelog: **https://github.com/gabime/spdlog/releases/tag/v1.5.0**

The above mentioned version is newly released by the upstream project and not yet available as a recipe. Please add this version.

### PR DESCRIPTION
**Pull request by KingKili**
Specify library name and version:  **spdlog/1.5.0** should close #671 

- [x] I've read the [guidelines](https://github.com/conan-io/conan-center-index/wiki#how-to-submit-a-pull-request) for contributing.
- [x] I've followed the [PEP8](https://www.python.org/dev/peps/pep-0008/) style guides for Python code in the recipes.
- [x] I've used the [latest](https://github.com/conan-io/conan/releases/latest) Conan client version.
- [x] I've tried at least one configuration locally with the
      [conan-center hook](https://github.com/conan-io/hooks.git) activated.

### CODE REVIEW

File: recipes/spdlog/all/conanfile.py [hunk 1]
```diff
@@ -45,7 +45,10 @@ def configure(self):
             raise ConanInvalidConfiguration("wchar is not yet supported under windows")
 
     def requirements(self):
-        self.requires("fmt/6.0.0")
+        if self.version == "1.5.0":
+            self.requires("fmt/6.1.2")
```
Comment (uilianries) [inline recipes/spdlog/all/conanfile.py]: fmt/6.1.2 does not it work with older spdlog versions?

Comment (KingKili) [reply recipes/spdlog/all/conanfile.py]: Yes you are right, I misread the changelog. I will change it.

File: recipes/spdlog/all/conanfile.py [hunk 2]
```diff
@@ -45,7 +45,7 @@ def configure(self):
             raise ConanInvalidConfiguration("wchar is not yet supported under windows")
 
     def requirements(self):
-        self.requires("fmt/6.0.0")
+            self.requires("fmt/6.0.0")
```
Comment (uilianries) [inline recipes/spdlog/all/conanfile.py]:
```suggestion
        self.requires("fmt/6.0.0")
```

Comment (uilianries) [reply recipes/spdlog/all/conanfile.py]: bad indentation, you have added more 4 spaces due the previous `if` which was removed. Please, fix it, than we are good! Regards!

Comment (KingKili) [reply recipes/spdlog/all/conanfile.py]: Definitely, that happens if you once edit files in browser ...

**Comment by uilianries** [comment]
@KingKili that patch can not be applied directly over 1.5.0.

**Comment by SSE4** [comment]
is this patch still needed for new release?

**Comment by uilianries** [comment]
@SSE4 Unfortunately yes, I've checked it now.

@KingKili Please, accept https://github.com/KingKili/conan-center-index/pull/2 which contains a fix.

**Pull request merged**